### ISSUE TITLE
Failure to compile cairo on macOS Sierra

### ISSUE THREAD

**Issue by andreabedini** [gtk2hs/gtk2hs#183]
I'm a newbie so I am not familiar with the whole stack. But I have this problem.

```
$ stack build cairo
[1 of 2] Compiling SetupWrapper     ( SetupWrapper.hs, /private/var/folders/rn/5myk4cmd49n0js1s8jr5977w0000gn/T/stack35188/cairo-0.13.1.1/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/setup/SetupWrapper.o )
[2 of 2] Compiling Main             ( /private/var/folders/rn/5myk4cmd49n0js1s8jr5977w0000gn/T/stack35188/cairo-0.13.1.1/Setup.hs, /private/var/folders/rn/5myk4cmd49n0js1s8jr5977w0000gn/T/stack35188/cairo-0.13.1.1/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/setup/Main.o )
Linking /private/var/folders/rn/5myk4cmd49n0js1s8jr5977w0000gn/T/stack35188/cairo-0.13.1.1/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/setup/setup ...
[1 of 2] Compiling Gtk2HsSetup      ( Gtk2HsSetup.hs, dist/setup-wrapper/Gtk2HsSetup.o )
[2 of 2] Compiling Main             ( SetupMain.hs, dist/setup-wrapper/Main.o )
Linking dist/setup-wrapper/setup ...
Configuring cairo-0.13.1.1...
Building cairo-0.13.1.1...
Preprocessing library cairo-0.13.1.1...
gtk2hsC2hs: Error in C header file.

/usr/include/stdio.h:133: (column 19) [FATAL] 
  >>> Syntax error!
  The symbol `_close' does not fit here.
```

line 133 of `stdio.h` is

```c
typedef struct __sFILE {
        ... omiss ...
        int     (* _Nullable _close)(void *);
        ... omiss ...
} FILE;
```

apparently `_Nullable` is defined in `/usr/include/sys/cdefs.h`.

**Comment by loisch** [comment]
I've got the same problem.  Any ideas?

**Comment by loisch** [comment]
I don't have macOS Sierra.  I only installed XCode 8.

**Comment by loisch** [comment]
I've found a workaround: I installed an old SDK from https://github.com/phracker/MacOSX-SDKs.

**Comment by timbod7** [comment]
@loisch How did you make the build process point to the old SDK?

**Comment by agrafix** [comment]
@timbod7 you download the SDK and then SymLink it. The only annoyance is that you will have to rebuild everything...

**Comment by agrafix** [comment]
Btw the bug is probably something related to: https://github.com/gtk2hs/gtk2hs/blob/b66eedc8ae9a9492f628a31fad065b95b226e911/tools/c2hs/c/CParser.y#L868

**Comment by hamishmack** [comment]
Sorry for the delay fixing this.  It turns out this error only happens when building from clean.  Fixed version of cairo (`cairo-0.13.3.1`) is now in hackage.  Just uploading fixes for the other Gtk2Hs packages now.

**Comment by timbod7** [comment]
@hamishmack Thanks for this. [Switching to the new packages][1] solved my issues with xcode 8.

[1]:https://github.com/timbod7/haskell-chart/commit/4e947d0d0197e85c7d66b46379f0bc58fd8b72b9
